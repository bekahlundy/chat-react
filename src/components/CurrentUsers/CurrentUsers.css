.CurrentUsers--title {
  font-size: 12px;
}

.CurrentUsers--current-user {
  background-color: #FFE8FC;
  font-weight: bolder;
}
