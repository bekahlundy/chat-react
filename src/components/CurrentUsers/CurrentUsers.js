import React, { Component } from "react";
import "./CurrentUsers.css"
export class CurrentUsers extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { currentUser, allUsers } = this.props;

    if (allUsers.length === 0 || !currentUser) {
      return <div className="CurrentUsers">something went wrong, try again</div>;
    }

    return (
      <div className="CurrentUsers--container">
        <div className="CurrentUsers--title">Current Users:</div>
        {allUsers.map(user => (
          <div key={user.id} className={ currentUser === user.user ? "CurrentUsers--current-user" : "CurrentUsers--user"}>{user.user}</div>
        ))}
      </div>
    );
  }
}
