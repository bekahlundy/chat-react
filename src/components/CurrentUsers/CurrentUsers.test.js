import React from "react";
import ReactDOM from "react-dom";
import { CurrentUsers } from "./CurrentUsers";
import { render, cleanup } from "react-testing-library";
import uuid from "uuid";

describe("CurrentUsers", () => {
  afterEach(() => {
    cleanup();
  });

  it("renders warning text if there is no current user", () => {
    const user = "";
    const users = [
      { user: "bekah", id: uuid.v4() },
      { user: "friend", id: uuid.v4() }
    ];

    const { container } = render(
      <CurrentUsers currentUser={user} allUsers={users} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders warning text if there are no users", () => {
    const user = "bekah";
    const users = [];

    const { container } = render(
      <CurrentUsers currentUser={user} allUsers={users} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders current users when there is both a user and a users array", () => {
    const user = "bekah";
    const users = [
      { user: "bekah", id: uuid.v4() },
      { user: "friend", id: uuid.v4() }
    ];

    const { container } = render(
      <CurrentUsers currentUser={user} allUsers={users} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
