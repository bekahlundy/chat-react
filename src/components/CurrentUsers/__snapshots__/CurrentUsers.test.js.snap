// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentUsers renders current users when there is both a user and a users array 1`] = `
<div
  class="CurrentUsers--container"
>
  <div
    class="CurrentUsers--title"
  >
    Current Users:
  </div>
  <div
    class="CurrentUsers--current-user"
  >
    bekah
  </div>
  <div
    class="CurrentUsers--user"
  >
    friend
  </div>
</div>
`;

exports[`CurrentUsers renders warning text if there are no users 1`] = `
<div
  class="CurrentUsers"
>
  something went wrong, try again
</div>
`;

exports[`CurrentUsers renders warning text if there is no current user 1`] = `
<div
  class="CurrentUsers"
>
  something went wrong, try again
</div>
`;
